python/taichi/_ti_module/__init__.py

```diff
@@ -55,6 +57,61 @@ def module_cppgen_impl(a):
     print(f"Module header is saved to: {Path(a.output).absolute()}")
 
 
+def module_build(parser: argparse.ArgumentParser):
+    """Build Taichi modules from python scripts."""
+    parser.add_argument(
+        "SOURCE", help="Path to the Taichi program source (Python script).")
+    parser.add_argument("-o",
+                        "--output",
+                        type=str,
+                        help="Output module path.",
+                        default=None)
+    parser.add_argument("-a",
+                        "--arch",
+                        type=str,
+                        help="Target architecture.",
+                        default="vulkan")
+    parser.set_defaults(func=module_build_impl)
+
+
+def module_build_impl(a):
+    source_path = a.SOURCE
+    module_path = a.output
+    arch = _ti_core.arch_from_name(a.arch)
+
+    source_path = Path(source_path)
+    assert source_path.name.endswith(".py"), "Source must be a Python script."
+    if module_path is None:
+        module_path = f"{source_path.name[:-3]}.tcm"
+    module_path = Path(module_path)
+
+    print()
+    print(f"Building Taichi module: {source_path}")
+
+    d = runpy.run_path(str(source_path), run_name="__main__")
+
+    required_caps = d["REQUIRED_CAPS"] if "REQUIRED_CAPS" in d else []
```

Comment:
would you mind adding a few sentences in PR summary describing how users should use `REQUIRED_CAPS` ?